core/tests/revert-test/tests/revert-and-restart-en.test.ts

```diff
@@ -0,0 +1,365 @@
+                break;
+            } catch (err) {
+                if (proc.exitCode != null) {
+                    assert.fail(`server failed to start, exitCode = ${proc.exitCode}`);
+                }
+                console.log('waiting for api endpoint');
+                await utils.sleep(1);
+            }
+        }
+        return new MainNode(tester, proc);
+    }
+
+    // Sends SIGINT to the main node process and waits for it to exit.
+    public async terminate(): Promise<void> {
+        this.proc.kill('SIGINT');
+        while (this.proc.exitCode == null) {
+            await utils.sleep(1);
+        }
+        expect(this.proc.exitCode).to.equal(0);
+    }
+}
+
+class ExtNode {
+    constructor(public tester: Tester, private proc: child_process.ChildProcess) {}
+
+    // Terminates all main node processes running.
+    public static async terminate_all() {
+        try {
+            await utils.exec('killall -INT zksync_external_node --wait');
+        } catch (err) {
+            console.log(`ignored error: ${err}`);
+        }
+    }
+
+    // Spawns an external node.
+    // If enable_consensus is set, the node will use consensus P2P network to fetch blocks.
+    public static async spawn(logs: fs.WriteStream, enable_consensus: boolean): Promise<ExtNode> {
+        let env = fetchEnv(ext_env);
+        console.log(`DATABASE_URL = ${env.DATABASE_URL}`);
+        let args = [];
+        if (enable_consensus) {
+            args.push('--enable-consensus');
+        }
+        let proc = spawn('./target/release/zksync_external_node', args, {
+            cwd: env.ZKSYNC_HOME,
+            stdio: [null, logs, logs],
+            env: env
+        });
+        // Wait until the node starts responding.
+        let tester: Tester = await Tester.init(env.EN_ETH_CLIENT_URL, `http://localhost:${env.EN_HTTP_PORT}`);
+        while (true) {
+            try {
+                await tester.syncWallet.provider.getBlockNumber();
+                break;
+            } catch (err) {
+                if (proc.exitCode != null) {
+                    assert.fail(`node failed to start, exitCode = ${proc.exitCode}`);
+                }
+                console.log('waiting for api endpoint');
+                await utils.sleep(1);
+            }
+        }
+        return new ExtNode(tester, proc);
+    }
+
+    // Sends SIGINT to the node process and waits for it to exit.
+    public async terminate(): Promise<void> {
+        this.proc.kill('SIGINT');
+        expect(await this.wait_for_exit()).to.equal(0);
+    }
+
+    // Waits for the node process to exit.
+    public async wait_for_exit(): Promise<number> {
+        while (this.proc.exitCode == null) {
+            await utils.sleep(1);
+        }
+        return this.proc.exitCode;
+    }
+}
+
+describe('Block reverting test', function () {
+    compileBinaries();
+    let main_logs: fs.WriteStream = fs.createWriteStream(main_logs_path, { flags: 'a' });
```

Comment:
Nit: In TS, `let` should be idiomatically used if the variable is reassigned (i.e., has shallow mutability); in all other cases, one should use `const`. Not a big deal, obviously.

Reply:
consted everythings